Index: src/main/java/org/mariadb/jdbc/MariaDbPoolDataSource.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/main/java/org/mariadb/jdbc/MariaDbPoolDataSource.java	(date 1555595923000)
+++ src/main/java/org/mariadb/jdbc/MariaDbPoolDataSource.java	(date 1555595923000)
@@ -31,6 +31,7 @@
 import java.util.Collections;
 import java.util.List;
 import java.util.Properties;
+import java.util.concurrent.locks.ReentrantLock;
 import java.util.logging.Logger;
 import javax.sql.DataSource;
 import javax.sql.XAConnection;
@@ -61,6 +62,7 @@
   private Integer maxIdleTime;
   private Boolean staticGlobal;
   private Integer poolValidMinDelay;
+  ReentrantLock lock = new ReentrantLock();

   /**
    * Constructor.
@@ -618,11 +620,18 @@
    *
    * @throws SQLException if connection string has error
    */
-  public synchronized void initialize() throws SQLException {
-    if (pool == null) {
-      initializeUrlParser();
-      pool = Pools.retrievePool(urlParser);
+  public void initialize() throws SQLException {
+    lock.lock();
+    try{
+      if (pool == null) {
+        initializeUrlParser();
+        pool = Pools.retrievePool(urlParser);
+      }
     }
+    finally {
+      lock.unlock();
+    }
+
   }

   /**
